### PR TITLE
Run tests in parallel in travis-ci

### PR DESCRIPTION
**Pull request by xnull**
## Overview

Description:
Run unit and integration tests on travis-ci in parallel which reduces total time to build and test a PR. 
Setup Maven wrapper to provide better automation and make builds more stable.

Why should this be merged: 
Improve tests performance in travis

Related issue(s) (if applicable): #<number>

## Checklist (Definition of Done):

- [x] There are no TODOs left in the code
- [x] [Coding conventions](https://github.com/CorfuDB/CorfuDB/wiki/Corfu-Style-Guidelines) (e.g. for logging, unit tests) have been followed
- [x] Change is covered by automated tests
- [x] Public API has Javadoc

### CODE REVIEW

File: .mvn/wrapper/MavenWrapperDownloader.java [hunk 1]
```diff
@@ -0,0 +1,110 @@
+/*
```
Comment (xnull) [inline .mvn/wrapper/MavenWrapperDownloader.java]: A review is not required, it's a middleware, it was generated by maven wrapper

File: .mvn/wrapper/maven-wrapper.properties [hunk 2]
```diff
@@ -0,0 +1 @@
+distributionUrl=https://repo.maven.apache.org/maven2/org/apache/maven/apache-maven/3.6.0/apache-maven-3.6.0-bin.zip
```
Comment (xnull) [inline .mvn/wrapper/maven-wrapper.properties]: A review is not required, it's a middleware, it was generated by maven wrapper

File: mvnw [hunk 3]
```diff
@@ -0,0 +1,286 @@
+#!/bin/sh
```
Comment (xnull) [inline mvnw]: A review is not required, it's a middleware, it is a maven wrapper

File: mvnw.cmd [hunk 4]
```diff
@@ -0,0 +1,161 @@
+@REM ----------------------------------------------------------------------------
```
Comment (xnull) [inline mvnw.cmd]: A review is not required, it's a middleware, it is a maven wrapper for windows

**Comment by no2chem** [comment]
Very nice, @xnull . I'll leave it to someone who has more recently touched the code to review it.

**Comment by Maithem** [comment]
Amazing. Thanks.

**Pull request merged**